subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -111,115 +111,192 @@ static int update_job_execution(struct mqtt_client *const client,
 	return ret;
 }
 
-
-static int aws_fota_on_publish_evt(struct mqtt_client *const client,
-				   const u8_t *topic,
-				   u32_t topic_len,
-				   u32_t payload_len)
+/**
+ * @brief Parsing an AWS IoT Job Execution response recived on $next/get MQTT
+ *	  topic or notify-next if it is a valid response the program state is
+ *	  updated.
+ *
+ * @param[in] client Connected MQTT client instance
+ * @param[in] payload_len Length of the payload going to be read out from the
+ *	      MQTT message.
+ *
+ * @return 1 If successful otherwise a negative error code is returned.
+ */
+static int get_job_execution(struct mqtt_client *const client,
+			     u32_t payload_len)
 {
-	int err;
+	int err = get_published_payload(client, payload_buf, payload_len);
 
-	LOG_INF("Received topic: %s", log_strdup(topic));
+	if (err) {
+		LOG_ERR("Error when getting the payload: %d", err);
+		return err;
+	}
+	/* Check if message received is a job. */
+	err = aws_fota_parse_DescribeJobExecution_rsp(payload_buf, payload_len,
+						      job_id, hostname,
+						      file_path,
+						     &execution_version_number);
+
+	if (err < 0) {
+		LOG_ERR("Error when parsing the json: %d", err);
+		return err;
+	} else  if (err == 1) {
+		LOG_DBG("Got only one field: %s", log_strdup(payload_buf));
+		LOG_INF("No queued jobs for this device");
+		return 1;
+	}
 
-	bool is_get_next_topic = aws_jobs_cmp(get_topic, topic, topic_len, "");
-	bool is_get_accepted =
-		aws_jobs_cmp(get_topic, topic, topic_len, "accepted");
-	bool is_notify_next_topic =
-		aws_jobs_cmp(notify_next_topic, topic, topic_len, "");
+	LOG_DBG("Job ID: %s", log_strdup(job_id));
+	LOG_DBG("hostname: %s", log_strdup(hostname));
+	LOG_DBG("file_path %s", log_strdup(file_path));
+	LOG_DBG("execution_version_number: %d ", execution_version_number);
+
+	/* Subscribe to update topic to recive feedback on whether an
+	 * update is accepted or not.
+	 */
+	err = aws_jobs_subscribe_topic_update(client, job_id, update_topic);
```

Comment:
Why is this done? (It's not mentioned in the function docs)

Reply:
This is done here as we don't have the `job_id` before we parse the response. I think you are correct in saying this should be a part of the function's documentation that it will subscribe to the update topic if an `Job Execution`containing a job description is returned.